app/models/user.rb

```diff
@@ -122,6 +122,13 @@ def suspend!
     OutOfBandSessionAccessor.new(unique_session_id).destroy if unique_session_id
     update!(suspended_at: Time.zone.now, unique_session_id: nil)
     analytics.user_suspended(success: true)
+    email_addresses.flat_map do |email_address|
+      digested_base_email = SuspendedEmail.generate_email_digest(email_address.email)
+      SuspendedEmail.create!(
+        digested_base_email: digested_base_email,
+        email_address: email_address,
+      )
```

Comment:
seems like a good case for a class-level `create_from_email_address` method:
```suggestion
      SuspendedEmail.create_from_email_adddress!(email_address)
```